test/Spaces/terrain_warp.jl

```diff
@@ -56,22 +56,42 @@ function warpedspace_2D(
     vert_face_space = Spaces.FaceFiniteDifferenceSpace(vertmesh)
 
     # Generate Horizontal Space
-    horzdomain = Domains.IntervalDomain(
-        Geometry.XPoint{FT}(xlim[1]),
-        Geometry.XPoint{FT}(xlim[2]);
-        periodic = true,
-    )
-    horzmesh = Meshes.IntervalMesh(horzdomain; nelems = helem)
-    horztopology = Topologies.IntervalTopology(horzmesh)
     quad = Spaces.Quadratures.GLL{npoly + 1}()
-    hspace = Spaces.SpectralElementSpace1D(horztopology, quad)
-
+    if ndims == 2
+        horzdomain = Domains.IntervalDomain(
+            Geometry.XPoint{FT}(xlim[1]),
+            Geometry.XPoint{FT}(xlim[2]);
+            periodic = true,
+        )
+        horzmesh = Meshes.IntervalMesh(horzdomain; nelems = helem)
+        horztopology = Topologies.IntervalTopology(horzmesh)
+        hspace = Spaces.SpectralElementSpace1D(horztopology, quad)
+    elseif ndims == 3
+        horzdomain = Domains.RectangleDomain(
+            Geometry.XPoint{FT}(xlim[1]) .. Geometry.XPoint{FT}(xlim[2]),
+            Geometry.YPoint{FT}(xlim[1]) .. Geometry.YPoint{FT}(xlim[2]),
+            x1periodic = true,
+            x2periodic = true,
+        )
+        horzmesh = Meshes.RectilinearMesh(horzdomain, helem, helem)
```

Comment:
`helem, helem` looks like a typo?

Reply:
I should rename these perhaps ; I just need a 3D space with the same number of elems in (x,y) here.

Reply:
If `RectilinearMesh` supports a single arg `helem` and assumes uniform number of elems that should fix this as well - I can check this.

Reply:
Ah, ok, it just looked odd at first glance, that's fine.

Reply:
Added commit with a comment to clarify that (x,y) coordinate directions have identical element numbers here.